src/ResourceFile.cc

```diff
@@ -669,6 +676,27 @@ static string format_template_bool(shared_ptr<const ResourceFile::TemplateEntry>
   return (value ? "true" : "false") + case_name_suffix;
 }
 
+static void format_list_item(deque<string>& lines, StringReader& r, const shared_ptr<const ResourceFile::TemplateEntry>& entry, size_t indent_level, size_t z) {
+  deque<string> temp_lines;
+  disassemble_from_template_inner(temp_lines, r, entry->list_entries, indent_level + 1);
+  
+  string item_prefix(indent_level * 2, ' ');
+  item_prefix += string_printf("%zu:", z);
+  
+  // When the inner template is a single line, prefix it with the array index.
+  // Otherwise put the array index on its own line
+  if (temp_lines.size() == 1) {
+    string last = temp_lines.back();
+    strip_leading_whitespace(last);
+    temp_lines.back() = item_prefix + " " + last;
+  } else {
+    lines.emplace_back(item_prefix);
+  }
+  for (const string& l : temp_lines) {
+    lines.emplace_back(l);
+  }

```

Comment:
I think it would be a bit cleaner if this was moved to the `else` block immediately above, and line 691 directly called `lines.emplace_back` instead.